src/filters/truncate.js

```diff
@@ -3,7 +3,7 @@
 angular.module('openshiftCommonUI')
   .filter('truncate', function() {
     return function(str, charLimit, useWordBoundary, newlineLimit) {
-      if (!str) {
+      if (typeof str !== 'string' || str.length <= charLimit) {
```

Comment:
If there is a hard newline limit like 1 line, this will return a string with newlines. Can we apply `useWordBoundary` only if the string is longer than limit, but still apply any newline truncation below?